Added evil-annoying-arrows

endrebak opened the pull request:
I created an evil-version of the annoying-arrows package (already in melpa) with the [blessing](https://github.com/magnars/annoying-arrows-mode.el/issues/5) of the [original author](https://github.com/magnars): 

```
I'm perfectly fine with a separate evil-annoying-arrows package,
or if you want to add a pull request that's good too. :)
```

I wanted to just update his package to work with evil (and truly tried), but evil is slightly quirky and it was a true pain in the ass. In the end I gave up on creating a version that worked for both vanilla and evil emacs.

All this package does for now is ring the bell if you use the arrow keys or hjkl too much. It also shows suggestions for more efficient commands to achieve the same movements.

purcell commented:
Thanks. You're implicitly using `remove-if` from `'cl`. Please instead use `cl-remove-if`, add `(require 'cl-lib)` and list `(cl-lib "0.5")` in the `Package-Requires` list.

endrebak commented:
Thanks for the swift review. Added your suggestions:  https://github.com/endrebak/evil-annoying-arrows/commit/fdf4dec1521a93e66be80ca46374b67bded27fb7

purcell commented:
Thanks. Squashed and applied in c76c862.

endrebak commented:
This still isn't up. Was there a problem with processing this package?

purcell commented:
Ah, right -- the file should definitely be renamed to `evil-annoying-arrows.el`. (`-mode` is usually redundant anyway: many things in Emacs are modes, and typically `-mode` is used only in the name of packages which provide major modes.)

purcell commented:
I'd still have expected the package to show up, though, with a blank package description. Let's see what happens when you rename the file.

endrebak commented:
Thanks for helping. I changed the filename and corrected the header which still referred to the file as `evil-annoying-arrows-mode.el`.

Where do I add a description? Can't find anything about it in the melpa docs.

And do I need to add another pull-request or should this take care of itself if I did everything correctly?

endrebak commented:
All is well: http://melpa.org/?utm_source=dlvr.it&utm_medium=twitter#/evil-annoying-arrows Thanks again for helping this rank newb.

purcell commented:
> Thanks for helping. I changed the filename and corrected the header which still referred to the file as evil-annoying-arrows-mode.el.

Great!

> 
> Where do I add a description? Can't find anything about it in the melpa docs. (This is just taken from the README it seems...)

It's the part after `---` on the first line.

And the `;;; Commentary:` section is used for the contents of the page, as you've probably seen now.

Glad it's all working. :-)
